Runtime/GameLiftCoreApi.cs

```diff
@@ -3,51 +3,51 @@
 
 #if !UNITY_SERVER
 
-using System;
 using System.Collections.Generic;
 using System.Linq;
 using System.Threading.Tasks;
+using AmazonGameLift.Editor;
+using AmazonGameLiftPlugin.Core;
 using AmazonGameLiftPlugin.Core.ApiGatewayManagement;
 using AmazonGameLiftPlugin.Core.ApiGatewayManagement.Models;
 using AmazonGameLiftPlugin.Core.Latency;
 using AmazonGameLiftPlugin.Core.Latency.Models;
 using AmazonGameLiftPlugin.Core.Shared;
 using AmazonGameLiftPlugin.Core.UserIdentityManagement;
 using AmazonGameLiftPlugin.Core.UserIdentityManagement.Models;
+using UnityEngine;
 
 namespace AmazonGameLift.Runtime
 {
     public class GameLiftCoreApi
     {
-        private readonly GameLiftConfiguration _configuration;
-        private readonly bool _isLocalMode;
+        private readonly GameLiftServerProcessConfiguration _configuration;
+        private readonly bool _isAnywhereMode;
+        private CoreApi _coreApi;
 
-        public GameLiftCoreApi(GameLiftConfiguration configuration)
+        protected GameLiftCoreApi(GameLiftServerProcessConfiguration configuration)
         {
             _configuration = configuration;
             _userIdentity = new UserIdentity(new AmazonCognitoIdentityWrapper(configuration.AwsRegion));
             _apiGateway = new ApiGateway(_userIdentity, new JwtTokenExpirationCheck(), new HttpClientWrapper());
-
-            if (_configuration.ApiGatewayEndpoint.StartsWith("https://localhost"))
-            {
-                throw new ArgumentException("Invalid configuration: https is not supported by GameLift Local.", nameof(configuration));
-            }
-
-            if (_configuration.ApiGatewayEndpoint.StartsWith("http://localhost"))
+            _coreApi = CoreApi.SharedInstance;
+            var fleetId = _coreApi.GetSetting("FleetId").Value;
+            var fleetLocation = _coreApi.GetSetting("FleetLocation").Value;
+            if (_configuration.IsGameLiftAnywhere) //TODO Review this when everything is merged into feature/anywhere.
             {
-                var gameLiftClientWrapper = new AmazonGameLiftClientWrapper(_configuration.ApiGatewayEndpoint);
-                _localGame = new LocalGameAdapter(gameLiftClientWrapper);
-                _isLocalMode = true;
+                var gameLiftClientWrapper = new AmazonGameLiftWrapper(_coreApi.RetrieveAwsCredentials(_coreApi.GetSetting(SettingsKeys.CurrentProfileName).Value)); 
```

Comment:
As pointed out in the other PR, it might be cleaner to just create the client here and pass it directly into the wrapper rather than pass credentials around